Update from auto-update/renku-core-2.0.0 (#2875)
SwissDataScienceCenter/renku: CHANGELOG.rst

.. _changelog:

0.22.0
------

> Renku ``0.22.0`` updates the core-service and CLI components to version 2.0, adding support for workflow files
> and requiring users to migrate their project metadata through the UI to version 10.
> It also phases out support for projects on metadata version 8, meaning those projects must be migrated to have
> full functionality in the UI.

Reviewer: ```suggestion
Renku ``0.22.0`` introduces the Renku Workflow File, a friendlier way to encode and run your data analysis pipelines on Renku. This yaml-based file is easy to write and execution is as easy as `renku run workflow.yml`. Add a workflow to your Renku project with our docs [here](https://renku.readthedocs.io/en/workflow-file-documentation/topic-guides/workflows/workflow-file.html)!

**Backwards-Compatibility Note**
This release also marks the update of the core-service and CLI components to version 2.0, which makes significant updates to Renku's internal metadata model. Users can migrate their project metadata to version 10 through the RenkuLab UI. **For projects on metadata version 8, this migration is required** to maintain full functionality in the UI.
```

Reviewer: 1. How can users determine which metadata version their project is on?
2. How exactly do users perform this migration in the UI? Do we have docs we can link to?

Reviewer 2: @lokijuhy this link will be broken as soon as we merge the docs PR: https://renku.readthedocs.io/en/workflow-file-documentation/topic-guides/workflows/workflow-file.html

Reviewer: @rokroskar Thanks! Fixed!

Reviewer 3: You can see the metadata version here (v9):
![image](https://user-images.githubusercontent.com/664486/210324909-f6b7e2f0-3d71-4593-a3e5-c4b6ff845871.png)
or do `renku migrate --check` and it'll tell you if a migration is needed

It would also show a button to click there that will perform the migration.

I don't think we have documentation for this for the UI, just for the [cli](https://renku.readthedocs.io/en/stable/renku-python/docs/reference/commands/migrate.html?highlight=migrate#module-renku.ui.cli.migrate)

User-Facing Changes
~~~~~~~~~~~~~~~~~~~

> **✨ Improvements**
> 
> * 🎨 **Renku CLI**: Add support for writing workflow files which can be executed directly, greatly improving the UX
>   around workflows
>   (`#3176 <https://github.com/SwissDataScienceCenter/renku-python/pull/3176>`_)

Reviewer: ```suggestion
**🌟 New Features**

* 📝 **Renku CLI**: Introduce the Renku Workflow File, a yaml-based file for defining Renku workflows (`docs <https://renku.readthedocs.io/en/workflow-file-documentation/topic-guides/workflows/workflow-file.html>`_)
  (`#3176 <https://github.com/SwissDataScienceCenter/renku-python/pull/3176>`_)

**✨ Improvements**

```

Reviewer 2: same comment as above - probably safer to use `latest`

> * 🎨 **Renku CLI**: Made `toil` the default workflow backend instead of `cwl`

Reviewer: ```suggestion
* ⚙️  **Renku CLI**: Made `toil` the default workflow backend instead of `cwl`
```
